File: src/Entity/CashRegister.php
<?php

namespace App\Entity;

use App\Repository\CashRegisterRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: CashRegisterRepository::class)]
class CashRegister
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name:'cashRegisterId')]
    private ?int $cashRegisterId = null;

    #[ORM\Column]
    private ?int $input = null;

    #[ORM\Column]
    private ?int $output = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $dateTransaction = null;

    #[ORM\Column(length: 30)]
    private ?string $type = null;

    #[ORM\Column]
    private ?float $somme = null;

    #[ORM\Column]
    private ?int $idEntity = null;

    public function getcashRegisterId(): ?int
    {
        return $this->cashRegisterId;
    }

    public function getInput(): ?int
    {
        return $this->input;
    }

    public function setInput(int $input): static
    {
        $this->input = $input;

        return $this;
    }

    public function getOutput(): ?int
    {
        return $this->output;
    }

    public function setOutput(int $output): static
    {
        $this->output = $output;

        return $this;
    }

    public function getDateTransaction(): ?\DateTimeInterface
    {
        return $this->dateTransaction;
    }

    public function setDateTransaction(\DateTimeInterface $dateTransaction): static
    {
        $this->dateTransaction = $dateTransaction;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;

        return $this;
    }

    public function getSomme(): ?float
    {
        return $this->somme;
    }

    public function setSomme(float $somme): static
    {
        $this->somme = $somme;

        return $this;
    }

    public function getIdEntity(): ?int
    {
        return $this->idEntity;
    }

    public function setIdEntity(?int $idEntity): static
    {
        $this->idEntity = $idEntity;

        return $this;
    }
}

File: src/main/java/entities/CashRegister.java
package entities;

import java.util.Date;
//import java.sql.Date;

public class CashRegister {
    private int cashRegisterId;
    private int input;
    private int output;
    private Date date_transaction;
    private String type;
    private Float somme;
    private int id_entity;

    public CashRegister() {
    }

    public CashRegister(int input, int output, Date date_transaction, String type, Float somme, int id_entity) {
        this.input = input;
        this.output = output;
        this.date_transaction = date_transaction;
        this.type = type;
        this.somme = somme;
        this.id_entity = id_entity;
    }

    public CashRegister(int cashRegisterId, int input, int output, Date date_transaction, String type, Float somme, int id_entity) {
        this.cashRegisterId = cashRegisterId;
        this.input = input;
        this.output = output;
        this.date_transaction = date_transaction;
        this.type = type;
        this.somme = somme;
        this.id_entity = id_entity;
    }

    public int getCashRegisterId() {
        return cashRegisterId;
    }

    public void setCashRegisterId(int cashRegisterId) {
        this.cashRegisterId = cashRegisterId;
    }

    public int getInput() {
        return input;
    }

    public void setInput(int input) {
        this.input = input;
    }

    public int getOutput() {
        return output;
    }

    public void setOutput(int output) {
        this.output = output;
    }

    public java.sql.Date getDate_transaction() {
        return new java.sql.Date(date_transaction.getTime());
    }

    public void setDate_transaction(Date date_transaction) {
        this.date_transaction = date_transaction;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Float getSomme() {
        return somme;
    }

    public void setSomme(Float somme) {
        this.somme = somme;
    }

    public int getId_entity() {
        return id_entity;
    }

    public void setId_entity(int id_entity) {
        this.id_entity = id_entity;
    }
}
